lib/libzfs/libzfs_util.c

```diff
@@ -1940,3 +1941,91 @@ zprop_iter(zprop_func func, void *cb, boolean_t show_all, boolean_t ordered,
 {
 	return (zprop_iter_common(func, cb, show_all, ordered, type));
 }
+
+static void
+nomem(void)
+{
+	(void) fprintf(stderr, gettext("internal error: out of memory\n"));
+	exit(1);
+}
+
+/*
+ * Utility function to guarantee malloc() success.
+ */
+
+static void *
+safe_malloc(size_t size)
+{
+	void *data;
+
+	if ((data = calloc(1, size)) == NULL)
+		nomem();
+
+	return (data);
+}
+
+#ifndef ZFS_VERSION_EXTRA
+#define ZFS_VERSION_EXTRA ""
+#endif
```

Comment:
Do we really need to introduce another type of version name, in addition to ZFS_META_VERSION, ZFS_META_RELEASE, ZFS_META_NAME, and ZFS_META_ALIAS?  Can we use one of those for this purpose instead?

Reply:
I didn't want to hijack an existing define, so I added one for those that would look at the code and use it if they wanted to.

Reply:
The `ZFS_META_RELEASE` field is intended for use by downstream packages, so I don't think you'd be hijacking an existing define.  You should need only to update the META file when building.